src/components/action-popover/action-popover-menu/action-popover-menu.component.tsx

```diff
@@ -108,54 +115,114 @@ const ActionPopoverMenu = React.forwardRef<
+      if (!previousOpenValue && firstFocusableItem !== -1)
+        setFocusIndex(firstFocusableItem);
+    });
+
     const onKeyDown = useCallback(
       (e) => {
         if (Events.isTabKey(e)) {
           e.preventDefault();
-          // TAB: close menu and allow focus to change to next focusable element
+          // TAB: close menu and allow focus to change to the next focusable element
           focusButton();
           setOpen(false);
         } else if (Events.isDownKey(e)) {
-          // DOWN: focus next item or first
+          // DOWN: focus on the next item or first non-disabled item
           e.preventDefault();
           e.stopPropagation();
-          const indexValue = focusIndex < items.length - 1 ? focusIndex + 1 : 0;
+          let indexValue = focusIndex + 1;
+          while (indexValue < items.length && isItemDisabled(indexValue)) {
+            indexValue += 1;
+          }
+          if (indexValue >= items.length) {
+            indexValue = firstFocusableItem;
+          }
           setFocusIndex(indexValue);
         } else if (Events.isUpKey(e)) {
-          // UP: focus previous item or last
+          // UP: focus on the previous item or last non-disabled item
           e.preventDefault();
           e.stopPropagation();
-          const indexValue = focusIndex > 0 ? focusIndex - 1 : items.length - 1;
+          let indexValue = focusIndex - 1;
+          while (
+            indexValue >= firstFocusableItem &&
+            isItemDisabled(indexValue)
+          ) {
+            indexValue -= 1;
+          }
+          if (indexValue < firstFocusableItem) {
+            indexValue = lastFocusableItem;
+          }
           setFocusIndex(indexValue);
         } else if (Events.isHomeKey(e)) {
-          // HOME: focus first item
+          // HOME: focus on the first non-disabled item
           e.preventDefault();
           e.stopPropagation();
-          setFocusIndex(0);
+          const indexValue = firstFocusableItem;
+          setFocusIndex(indexValue);
         } else if (Events.isEndKey(e)) {
-          // END: focus last item
+          // END: focus on the last non-disabled item
           e.preventDefault();
           e.stopPropagation();
-          setFocusIndex(items.length - 1);
+          const indexValue = lastFocusableItem;
+          setFocusIndex(indexValue);
         } else if (e.key.length === 1) {
-          // any printable character: focus the next item on the list that starts with that character
-          // selection should wrap to the start of the list
+          // Any printable character: focus on the next non-disabled item on the list that starts with that character
+          // Selection should wrap to the start of the list
           e.stopPropagation();
           let firstMatch: number | undefined;
           let nextMatch: number | undefined;
-          React.Children.forEach(items, (child, index) => {
+          for (let index = 0; index < items.length; index++) {
```

Comment:
Suggestion/nit-pick (non-blocking): we could probably use the forEach array function here if you want 

```
items.forEach((item, index) => {
            if (
              React.isValidElement(item) &&
              !isItemDisabled(index) &&
              item.props.children.toLowerCase().startsWith(e.key.toLowerCase())
            ) {
              // istanbul ignore else
              if (firstMatch === undefined) {
                firstMatch = index;
              }
              if (index > focusIndex && nextMatch === undefined) {
                nextMatch = index;
              }
            }
          })

Reply:
Good suggestion. I'll give it a whirl and see if it's all good 😄